### DISCUSSION TITLE
Blank page when deploying React app on GitHub Pages

### DISCUSSION THREAD

**Discussion by loicmaes** [community/community#120942] [Programming Help]
### Select Topic Area

Question

### Body

Hi,  
In my development environment, my React app is rendering fine but when building it and deploying it on GitHub pages, i'm facing to a blank page...

You can try it by yourself [here](https://loicmaes.github.io/oc-p5-kasa) or view the source code [here](https://github.com/loicmaes/oc-p5-kasa).

This is an extract of my code (only main and router file):

`main.tsx`
```tsx
import React from 'react'
import ReactDOM from 'react-dom/client'
import {RouterProvider} from 'react-router-dom'
import { router } from './router'
import './reset.sass'
import './pages/page.sass'

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
)
```

`router.tsx`
```tsx
import {createBrowserRouter, RouteObject} from 'react-router-dom'
import Home from '../pages/home'
import NotFound from '../pages/notFound'
import Layout from "../layouts/Layout.tsx"
import RentViewer from '../pages/rentViewer'
import About from "../pages/about";

export const routes: RouteObject[] = [
  {
    path: '/',
    element: <Layout />,
    errorElement: <NotFound />,
    children: [
      {
        path: '',
        element: <Home />
      },
      {
        path: 'about',
        element: <About />
      },
      {
        path: 'rent/:id',
        element: <RentViewer />
      }
    ]
  }
]

export const router = createBrowserRouter(routes)
```

Can somebody help me?

Thanks

**Comment by LiteBrite82** [comment]
looking at your code; I think this is a coding issue more than a pages issue. Pages is working; it is displaying the content of index.html. Though there are errors on the page. I suspect it has to do with how you are linking your files in your code. 

did you run a build? 

I'm going to move this discussion over to Programming Help; we may get more eyes on it over there.

**Comment by minions514** [comment]
I think you should try to build production mode on your local first.
